### ISSUE TITLE
Segmentation fault on empty study simulation (v8.1)

### ISSUE THREAD

**Issue by laurent-laporte-pro** [AntaresSimulatorTeam/Antares_Simulator#1245]
**Description**

After upgrading an empty study from v8.0 to v8.1 on Antares Web (v2.13.1), I launched a simulation using Antares
Solver v8.1.1 and I got a Segmentation fault.

**How to Reproduce**

Using Antares Web v2.13.1:

- Create a new study in version v8.0
- Upgrade the study to version v8.1
- Launch a simulation (using default parameters)

**Expected behavior**

- a "normal" error saying that there is no area defined or something analogous.

**Screenshots and additional files**

STDOUT:

```shell
Exporting study...
Extracting study 19f67370-c7be-4087-815b-d9963f1951d2
Study extracted
Checking study version...
Submitting study to slurm launcher
Study submitted
[Wed Apr  5 15:20:42 2023][noname][infos] Copyright 2007-2018 RTE  - Authors: The Antares_Simulator Team 
[…]
[Wed Apr  5 15:20:42 2023][solver][1;37m[1m[check][0m [1;37m[1mAntares Solver v8.1.1 (RTE France)[0m
[Wed Apr  5 15:20:42 2023][solver][infos]   :: built for 64-bit architectures, GNU/Linux, 36 cpu(s)
[Wed Apr  5 15:20:42 2023][solver][infos]   :: hostname = pf9socalant04.applispfref.sipfref.local
[Wed Apr  5 15:20:42 2023][solver][infos] 
[Wed Apr  5 15:20:42 2023][solver][infos]   :: from /applis/antares/solver/8.1.1
[Wed Apr  5 15:20:42 2023][solver][infos]   :: log filename: /scratch/antares_16756/f22abcba-6769-455c-82ba-9bc40bbf0260/logs/solver-20230405-152042.log
[Wed Apr  5 15:20:42 2023][solver][infos] 
[Wed Apr  5 15:20:42 2023][solver][0;32m[notic][0m Preparing issue-1209...
[Wed Apr  5 15:20:42 2023][solver][infos]   detected version: 810
[Wed Apr  5 15:20:42 2023][solver][infos]   from `/scratch/antares_16756/f22abcba-6769-455c-82ba-9bc40bbf0260`
[Wed Apr  5 15:20:42 2023][solver][infos] 
[Wed Apr  5 15:20:42 2023][solver][infos]   simulation mode: Economy
[Wed Apr  5 15:20:42 2023][solver][infos]   simplex optimization range: week
[Wed Apr  5 15:20:42 2023][solver][infos] Aggregate renewables were chosen. Output will be disabled for renewable clusters.
[Wed Apr  5 15:20:42 2023][solver][infos]   :: ignoring export mps
[Wed Apr  5 15:20:42 2023][solver][infos]   :: ignoring export structure
[Wed Apr  5 15:20:42 2023][solver][infos]   Output folder : /scratch/antares_16756/f22abcba-6769-455c-82ba-9bc40bbf0260/output/20230405-1520eco
[Wed Apr  5 15:20:42 2023][solver][infos] Loading the list of areas...
[Wed Apr  5 15:20:42 2023][solver][infos]   Loading the area list from `/scratch/antares_16756/f22abcba-6769-455c-82ba-9bc40bbf0260/input/areas/list.txt`
[Wed Apr  5 15:20:42 2023][solver][infos]   No area found
[Wed Apr  5 15:20:42 2023][solver][infos] Loading global hydro data...
[Wed Apr  5 15:20:42 2023][solver][infos] Loading thermal clusters...
[Wed Apr  5 15:20:42 2023][solver][infos] Loading correlation matrices...
[Wed Apr  5 15:20:42 2023][solver][infos] 
[Wed Apr  5 15:20:42 2023][solver][infos] Loading constraints...
[Wed Apr  5 15:20:42 2023][solver][infos] No binding constraint found
[Wed Apr  5 15:20:42 2023][solver][infos] 
[Wed Apr  5 15:20:42 2023][solver][infos] Loading sets of areas...
[Wed Apr  5 15:20:42 2023][solver][infos]    found `all areas` (0 item, no output)
[Wed Apr  5 15:20:42 2023][solver][infos]  Elapsed time: Study loading: 4ms
[Wed Apr  5 15:20:42 2023][solver][infos] [statistics] disk: read: 2 ko, written: 0 ko
[Wed Apr  5 15:20:42 2023][solver][infos] [statistics] network: read: 0 ko, written: 0 ko
[Wed Apr  5 15:20:42 2023][solver][infos] The study is loaded.
[Wed Apr  5 15:20:42 2023][solver][infos] [UI] Display messages: Off
[Wed Apr  5 15:20:42 2023][solver][infos] [UI] Quitting the solver gracefully
[Wed Apr  5 15:20:42 2023][solver][infos]  Writing log file: /scratch/antares_16756/f22abcba-6769-455c-82ba-9bc40bbf0260/output/20230405-1520eco/simulation.log

Simulation failed, output results are not available
```

STDERR:

```shell
[Wed Apr  5 15:20:42 2023][solver][0;31m[fatal][0m no area found
srun: error: pf9socalant04: task 0: Segmentation fault (core dumped)
```

**OS and version**

- OS: Linux
- Antares Simulator version: v8.1.1

**Additional context**

- The study is empty
- Study UUDI: '19f67370-c7be-4087-815b-d9963f1951d2'

**Comment by laurent-laporte-pro** [comment]
When the study is upgraded to v8.2 and launched with Antares Solver v8.2.3, there is no segmentation fault.

We have the following error message instead:

```shell
srun: error: pf9socalant04: task 0: Exited with exit code 1
[Wed Apr  5 15:48:02 2023][solver][0;31m[error][0m No area found. A valid study contains contains at least one.
[Wed Apr  5 15:48:02 2023][solver][0;31m[error][0m Aborting now. See logs for more details
```

This message is the correct behaviour.

**Comment by flomnes** [comment]
This seems to be related to a [segfault occurring when no INI file is present](https://antares-simulator.readthedocs.io/en/latest/CHANGELOG/#bug-fixes_5) (Linux only), it was fixed in v8.2.1

**Comment by flomnes** [comment]
See #606
